src/main/java/executor/service/service/impl/DIFactory.java

```diff
@@ -0,0 +1,79 @@
+package executor.service.service.impl;
+
+import executor.service.config.bean.BeanConfig;
+import executor.service.config.properties.PropertiesConfig;
+import executor.service.model.WebDriverConfig;
+import executor.service.service.*;
+import executor.service.service.impl.listener.*;
+import executor.service.service.impl.parallel.ParallelFlowExecutorServiceImpl;
+import executor.service.service.impl.parallel.TasksFactoryImpl;
+import executor.service.service.impl.stepExecution.StepExecutionClickCssImpl;
+import executor.service.service.impl.stepExecution.StepExecutionClickXpathImpl;
+import executor.service.service.impl.stepExecution.StepExecutionSleepImpl;
+import executor.service.service.impl.webDriver.WebDriverInitializerImpl;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+
+/**
+ * Factory for creating objects with dependency injection.
+ * This class is intended for creating objects of various classes while considering their dependencies.
+ * It utilizes caching to avoid creating the same objects multiple times.
+ * Example usage:
+ * <pre>
+ * DIFactory factory = new DIFactory();
+ * ParallelFlowExecutorService parallelExecutor = factory.createObject(ParallelFlowExecutorService.class);
+ * // Now you have an instance of ParallelFlowExecutorService with properly configured dependencies.
+ * </pre>
+ *
+ * @author Dima Silenko, Oleksandr Tuleninov.
+ * @version 01
+ */
+public class DIFactory implements ObjectFactory {
+
```

Comment:
I try to do something like this, but doesn't work:

DIFactory diFactory = new DIFactory();
        //ParallelFlowExecutorService service = diFactory.createObject(ParallelFlowExecutorService.class);
        //service.execute();

        ItemQueue itemQueue = new ItemQueue();
        ScenarioSourceListener listener = new ScenarioSourceListenerImpl(diFactory.createObject(ScenarioProvider.class)
        ,diFactory.createObject(PropertiesConfig.class));
        listener.execute(createHandler(itemQueue::putItem));
        listener.execute(createHandler(itemQueue::putItem));
        listener.execute(createHandler(itemQueue::putItem));
        listener.execute(createHandler(itemQueue::putItem));
        listener.execute(createHandler(itemQueue::putItem));
        listener.execute(createHandler(itemQueue::putItem));
        listener.execute(createHandler(itemQueue::putItem));

Reply:
Please show me a place where this makes sense. All dependencies are created when ParallelFlowExecutorServiceImpl is created and ParallelFlowExecutorServiceImpl is added to the cache, ParallelFlowExecutorServiceImpl is called in App.class, based on this, it makes no sense to add all dependencies to the cache.

Reply:
I want to try it run in such case, why I can not do this? I'm interesting, how flux wouldn't to read every time when I call listener execute value from properties

Reply:
![image](https://github.com/GFLCourses5/worker/assets/140727960/32b52fb3-68d3-410c-a231-34d7055b8f28)

Why are you doing this, in the program code this happens only once, look at the code carefully.

Reply:
Because I interest how many times flux will read delay from properties